### PR TITLE
Create 122_profit

### PR DESCRIPTION
**Pull request by jsingh41**
@sb2nov

### CODE REVIEW

File: 122_profit [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        :type prices: List[int]
+        :rtype: int
+        """
+        buy = False
+        profit = 0
+        for i in range(0,len(prices) - 1):
+            if prices[i] < prices[i + 1]:
+                buy = True
+                profit = profit + prices[i + 1] - prices[i]
+            elif prices[i] > prices[i + 1]:
```
Comment (sb2nov) [inline 122_profit]: what does the else here do? why do you need the buy flag?

File: 122_profit [hunk 2]
```diff
@@ -0,0 +1,30 @@
+"""
```
Comment (sb2nov) [inline 122_profit]: File should have .py at the end

**Comment by jsingh41** [comment]
buy keeps track of whether we have currently bought the stock or not. As you can't purchase two stocks at the same time, you have to sell your stock before buying again.

**Comment by jsingh41** [comment]
So the logic is:
To make as much profit as possible, compare the current and next value. If the next value > current value, that's profit
- If you aren't holding any stock right now, buy the stock so that you earn the profit of next_value - current_value
- If you are holding the stock, keep holding the stock

If the next value is less than current value:
- If you are holding the stock, sell it, and buy it later if possible
- If you are not holding the stock, don't do anything

**Comment by sb2nov** [comment]
```
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        profit = 0
        for i in range(0,len(prices) - 1):
            if prices[i] < prices[i + 1]:
                profit = profit + prices[i + 1] - prices[i]
        return profit
```

Will this not work as you're not using the buy flag in your profit adding clause anyway.

**Pull request merged**